node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -3747,6 +3939,39 @@ mod tests {
         assert_eq!(dest_db.node_by_key(node_f.public_key()), None);
     }
 
+    fn fix_last_update_nodes(
+        expected_db: &mut NeighborhoodDatabase,
+        dest_db: &NeighborhoodDatabase,
+        root_node: &NodeRecord,
+    ) {
+        expected_db
+            .node_by_key_mut(root_node.public_key())
+            .unwrap()
+            .metadata
+            .last_update = dest_db
+            .node_by_key(root_node.public_key())
+            .unwrap()
+            .metadata
+            .last_update;
```

Comment:
```Rust
       expected_db
            .node_by_key_mut(root_node.public_key())
            .unwrap()
            .metadata
            .last_update = dest_db
            .node_by_key(root_node.public_key())
            .unwrap()
            .metadata
            .last_update;
```

This block of code can be reused.
Assign it to a closure or make it a function, both with a single argument - the pub_key of the Node which it applies to) and use it up here, and then down there  with the `for_each()`.